rclrs/src/context.rs

```diff
@@ -118,6 +127,47 @@ impl Context {
     }
 }
 
+/// Returns non-ROS arguments held by context.
+///
+/// This function should be called after rcl_init. `rcl_arguments` should be `global_arguments`
+/// field of initialized context. `args` is array of input arguments passed to node.
+fn get_non_ros_arguments(
+    rcl_arguments: *const rcl_arguments_t,
+    args: &[*const c_char],
+) -> Result<Vec<String>, RclrsError> {
+    // SAFETY: In case of error -1 will be returned and handled a line below.
+    let args_status = unsafe { rcl_arguments_get_count_unparsed(rcl_arguments) };
+    if args_status == -1 {
+        return Err(RclrsError::RclError {
+            code: InvalidArgument,
+            msg: None,
+        });
+    }
+    // SAFETY: All possible negative args_status values handled above.
+    let args_num = args_status as usize;
+    if args_num == 0 {
+        return Ok(Vec::new());
+    }
+    let mut out_ptr: *mut i32 = null_mut();
+    unsafe {
+        // SAFETY: No preconditions for this function.
+        let allocator = rcutils_get_default_allocator();
+        // SAFETY: Error will be returned in case of broken rcl_arguments, no precondition for the rest.
+        rcl_arguments_get_unparsed(rcl_arguments, allocator, &mut out_ptr).ok()?;
```

Comment:
This safety comment needs to be expanded as to why `out_ptr` can be safely used. Specifically, in the event of no arguments, `out_ptr` will be null, though an error will not be raised.

Reply:
Actually there shouldn't be a situation when there's no arguments as stated [here](https://github.com/ros2/rcl/blob/foxy/rcl/include/rcl/arguments.h#L150), but yeah in this code such situation should be considered.